api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java

```diff
@@ -654,4 +657,12 @@ public int stopAllUserAppsAsService(String userEmail) {
 
     return results.size();
   }
+
+  @Override
+  public List<LeonardoListPersistentDiskResponse> listDisksByProjectAsService(
+      String googleProject) {
+    DisksApi disksApi = serviceDiskApiProvider.get();
+    return leonardoRetryHandler.run(
+        (context) -> disksApi.listDisksByProject(googleProject, null, false, null, null));
```

Comment:
I think we still need the other two: 
https://github.com/all-of-us/workbench/blob/56cfb5d6b68ea161fb6d73b8057ee03204e4898f/api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java#L448L449

Then group by creators, and types, otherwise, AppType and isGce will be null:
https://github.com/all-of-us/workbench/blob/cb53153ba72b43a376c21f608b982f829ae6ec55/api/src/main/java/org/pmiops/workbench/api/DisksController.java#L79

Reply:
I added includeDeleted and LeonardoLabelHelper.LEONARDO_DISK_LABEL_KEYS.

I do not understand findTheMostRecentActiveDisks is supposed to do. Wouldn't it filter out duplicate persistent disks if a user or workspace has them.

Would you mind elaborating why you would call findTheMostRecentActiveDisks?

Reply:
Include a test of a user with two distinct persistent disks of a particular app type. Maybe include comment to specify the need for it.

Reply:
ah, in your case, we do want to show all disks that a user has, so we don't actually want `findTheMostRecentActiveDisks`
Sorry for misleading.

Reply:
No worries. Thanks for clarifying.